util/build_docs.py

```diff
@@ -196,6 +196,23 @@ def generate_apt_reqs():
     with open(str(apt_cmd_path), mode='w') as fout:
         fout.write(apt_cmd)
 
+def generate_tool_versions():
+    """Generate an tool version number requirement from tool_requirements.py
+
+    The version number per tool will be saved in outdir-generated/version_$TOOL_NAME.txt
+    """
+
+    # Populate __TOOL_REQUIREMENTS__
+    requirements_file = str(SRCTREE_TOP.joinpath("tool_requirements.py"))
+    exec(open(requirements_file).read(), globals())
+
+    # And then write a version file for every tool.
+    for tool in __TOOL_REQUIREMENTS__:
+        version_path = config["outdir-generated"].joinpath('version_' + tool + '.txt')
```

Comment:
```suggestion
        version_path = config["outdir-generated"] / 'version_' + tool + '.txt'
```

Reply:
Thanks for your feedback @imphil . Your suggestion does not work as the first part is of type `PosixPath` and the second part is `str`. We are using `joinpath()` in several places in this file. I thus suggest to leave it.

Reply:
'/' also works with strings, but operator precedence requires parentheses around the string concatenation with `+`. So `config["outdir-generated"] / ('version_' + tool + '.txt')` or `config["outdir-generated"] / 'version_%s.txt' % tool` should do the trick.

No need to change that now, but to keep in mind for future changes.